protocol/synthetix/contracts/modules/core/VaultModule.sol

```diff
@@ -93,73 +111,160 @@ contract VaultModule is IVaultModule {
-        } else {
-            Pool.loadExisting(poolId).requireMinDelegationTimeElapsed(
-                vault.currentEpoch().lastDelegationTime[accountId]
-            );
         }
 
-        // Update the account's position for the given pool and collateral type,
-        // Note: This will trigger an update in the entire debt distribution chain.
-        uint256 collateralPrice = _updatePosition(
-            accountId,
-            poolId,
-            collateralType,
-            newCollateralAmountD18,
-            currentCollateralAmount,
-            leverage
-        );
-
-        _updateAccountCollateralPools(
+        // Create a new delegation intent.
+        intentId = DelegationIntent.nextId();
+        DelegationIntent.Data storage intent = DelegationIntent.load(intentId);
+        intent.id = intentId;
+        intent.accountId = accountId;
+        intent.poolId = poolId;
+        intent.collateralType = collateralType;
+        intent.deltaCollateralAmountD18 = deltaCollateralAmountD18;
+        intent.leverage = leverage;
+        intent.declarationTime = block.timestamp.to32();
+
+        // Add intent to the account's delegation intents.
+        AccountDelegationIntents.getValid(intent.accountId).addIntent(intent);
+
+        // emit an event
+        emit DelegationIntentDeclared(
             accountId,
             poolId,
             collateralType,
-            newCollateralAmountD18 > 0
+            deltaCollateralAmountD18,
+            leverage,
+            intentId,
+            intent.declarationTime,
+            intent.processingStartTime(),
+            intent.processingEndTime(),
+            ERC2771Context._msgSender()
         );
+    }
 
-        // If decreasing the delegated collateral amount,
-        // check the account's collateralization ratio.
-        // Note: This is the best time to do so since the user's debt and the collateral's price have both been updated.
-        if (newCollateralAmountD18 < currentCollateralAmount) {
-            int256 debt = vault.currentEpoch().consolidatedDebtAmountsD18[accountId];
-
-            uint256 minIssuanceRatioD18 = Pool
-                .loadExisting(poolId)
-                .collateralConfigurations[collateralType]
-                .issuanceRatioD18;
+    /**
+     * @inheritdoc IVaultModule
+     */
+    function processIntentToDelegateCollateralByIntents(
+        uint128 accountId,
+        uint256[] memory intentIds
+    ) public override {
+        FeatureFlag.ensureAccessToFeature(_DELEGATE_FEATURE_FLAG);
+        for (uint256 i = 0; i < intentIds.length; i++) {
+            DelegationIntent.Data storage intent = DelegationIntent.load(intentIds[i]);
+            if (!intent.isExecutable()) {
+                // Remove the intent.
+                if (intent.intentExpired()) {
+                    AccountDelegationIntents.getValid(accountId).removeIntent(intent);
+                    emit DelegationIntentRemoved(
```

Comment:
If is not executable, it will skip it and emit an event to indicate it was Skipped.

And, if it's expired it will also remove the intent and emit an event to indicate it was removed. 

So, you'll only get the skip event (and not the removed) if the time is too soon to execute this intent, that is, initial delay not reached

Reply:
if due to clarity, you prefer I move Skipped event before this if, I'll do it.